src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -565,25 +562,27 @@ ERaw m f ^. field
             persistFieldDef field
     idFieldValue =
         case getEntityKeyFields ed of
-            idField :| [] ->
-                ERaw noMeta $ \_ info -> (dot info idField, [])
+            idField :| [] -> ERaw noMeta $ \_ info -> (viewFieldBuilder ent info idField, [])
 
             idFields ->
-                let renderedFields info = dot info <$> NEL.toList idFields
+                let renderedFields info = viewFieldBuilder ent info <$> NEL.toList idFields
                 in ERaw noMeta{ sqlExprMetaCompositeFields = Just renderedFields} $
                     \p info -> (parensM p $ uncommas $ renderedFields info, [])
 
     ed = entityDef $ getEntityVal (Proxy :: Proxy (SqlExpr (Entity val)))
 
-    dot info fieldDef' =
-        sourceIdent info <> "." <> fieldIdent
-      where
-        sourceIdent = fmap fst $ f Never
-        fieldIdent
-            | Just baseI <- sqlExprMetaAlias m =
-                useIdent info $ aliasedEntityColumnIdent baseI fieldDef'
-            | otherwise =
-                fromDBName info (coerce $ fieldDB fieldDef')
+viewFieldBuilder :: SqlExpr (Entity val) -> IdentInfo -> FieldDef -> TLB.Builder
+viewFieldBuilder (ERaw m f) info fieldDef 
+    | Just alias <- sqlExprMetaAlias m =
+        fst (f Never info) <> ("." <> useIdent info (aliasedEntityColumnIdent alias fieldDef))
+    | otherwise = sourceIdent info <> "." <> fieldIdent
+  where
+    sourceIdent = fst <$> f Never
+    fieldIdent 
+        | Just baseI <- sqlExprMetaAlias m =
+            useIdent info $ aliasedEntityColumnIdent baseI fieldDef
+        | otherwise =
+            fromDBName info (coerce $ fieldDB fieldDef)
```

Comment:
```suggestion
viewFieldBuilder :: SqlExpr (Entity val) -> IdentInfo -> FieldDef -> TLB.Builder
viewFieldBuilder (ERaw m f) info fieldDef =
    sourceIdent info <> "." <> fieldIdent
  where
    sourceIdent = fst <$> f Never
    fieldIdent 
        | Just baseI <- sqlExprMetaAlias m =
            useIdent info $ aliasedEntityColumnIdent baseI fieldDef
        | otherwise =
            fromDBName info (coerce $ fieldDB fieldDef)
```